### ISSUE TITLE
Provide the bot with full information about the game state.

### ISSUE THREAD

**Issue by DomNomNom** [drssoccer55/RLBot#33]
There are a few minor things missing in the game_tick_packet which prevents bots from taking in a packet and determining the best play from just that packet. Currently it's impossible to predict whether setting bJump to true will cause the car to take action.

- To be able to double-jump as late as possible, PlayerInfo should track how long the car has been in the air, for the purpose of determining whether we are close to the [magic 1.5s timer](https://www.reddit.com/r/RocketLeague/comments/6h6hq3/second_jump_timer_aerial_question/).
- To be able to double-jump as quickly as possible, PlayerInfo should track whether the jump input is currently being pressed. ie. we should not jump the frame after we jump as otherwise it'll look like the button is being held down to the game engine.

Counter argument to "This is giving the bot an advantage over humans": 
Yes, but the current architecture already does give extra information which is not available to humans. eg. info about things that are not in the field of view and current velocities (not just positions).

I think this change should be done soon after this tournament round is over to provide the community with a stable API for most of a season.

**Comment by Trollgeir** [comment]
Also to the counter argument; These features are already being implemented by custom Agent scripts anyways.

It makes sense to include commonalities in PlayerInfo as long as the calculations are inexpensive.

**Comment by tarehart** [comment]
I agree with adding these features, and the jump timer situation is already on the dll backlog (which we should probably make public if possible).

Dom I think you're well aware of these specifics, but to the casual reader I want to point out that having this information from the packet does not remove all the guesswork. The packet can tell you that you still have a jump, but it may already be too late due to latency in the round trip.

Likewise, you'll need to take a risk if you want the fastest possible button taps, because waiting for confirmation will slow you down.

We should remember to document these caveats when the feature is added.
